### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by kermitt2** [kermitt2/delft#1]
Retraining a model leads to different evaluation due to different random seed. For instance NER f1 score with BidLSTM-CRF can go from 90.22 to 91.07 (on `eng.testb`).

* adding for Keras reproducibility, the classical:

```python
import numpy as np
np.random.seed(1)
```

* adding for tensorflow reproducibility:

```python
from tensorflow import set_random_seed
set_random_seed(1)
```

(does it add a tensorflow backend dependency in Keras?)

* test ... 

* make random seed optional

**Comment by kermitt2** [comment]
So we simply systematically averaging results over 10 training runs to get something stable.
